From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: buildroot <buildroot@buildroot.org>
Date: Thu, 20 Feb 2026 23:52:00 +0000
Subject: [PATCH] Fix Python 3.12 compatibility in wscript

Python 3.12 no longer accepts type='string' in argparse.
Change to type=str to fix compatibility.

Signed-off-by: buildroot <buildroot@buildroot.org>
---
 wscript | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/wscript b/wscript
index 1234567..abcdefg 100644
--- a/wscript
+++ b/wscript
@@ -935,10 +935,13 @@ def options(opt):
     for ident, option in sorted(build_options.items()):
         if option['action'] == 'store':
             default = option['default']
+            opt_type = option.get('type')
+            # Python 3.12+ requires type=str instead of type='string'
+            if opt_type == 'string':
+                opt_type = str
             group.add_option('--{0}'.format(ident),
                              action = option['action'],
                              dest   = ident,
-                             type   = option['type'])
+                             type   = opt_type)
 
     group.add_option('--lua',
                      action = 'store',
-- 
2.45.2

